webtau-config/src/main/java/org/testingisdocumenting/webtau/cfg/WebTauCoreConfigHandler.java

```diff
@@ -37,5 +37,47 @@
     public void onAfterCreate(WebTauConfig cfg) {
         SystemTimerConfig.setWaitTimeout(cfg.waitTimeout());
         DocumentationArtifactsLocation.setRoot(cfg.getDocArtifactsPath());
+
+        setProxy(cfg);
+    }
+
+    private void setProxy(WebTauConfig cfg) {
+        setProxy("http", cfg.getHttpProxyConfigValue());
+        setProxy("https", cfg.getHttpsProxyConfigValue());
+        setNoProxy(cfg);
+    }
+
+    private void setProxy(String httpPrefix, ConfigValue configValue) {
+        if (configValue.isDefault()) {
+            return;
+        }
+
+        HostPort hostPort = new HostPort(configValue.getAsString());
+        System.setProperty(httpPrefix + "." + "proxyHost", hostPort.host);
```

Comment:
Can you remind me exactly when the config handling would happen in a Junit test?  IIRC the proxy setup is essentially a lazily initialised singleton so if there was any way for someone to invoke some code, prior to cfg handling, which ends up initialising http proxies, this property setting here won't have any effect.